var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -235,6 +235,9 @@ def define_feature(key, variant=None):
         for k in features:
             define("FEATURE_" + k, True)
 
+        if "~opengl" in spec:
+            args.append(self.define("INPUT_opengl", "no"))
```

Comment:
```suggestion
            args.append(self.define("INPUT_opengl", False))
```
Assuming a cmake Boolean

Reply:
Unfortunately this does not work (I just tested it).
In `src/gui/configure.cmake` the test that checks that is:
```
qt_configure_add_report_entry(
    TYPE ERROR
    MESSAGE "The OpenGL functionality tests failed! You might need to modify the OpenGL package search path by setting the OpenGL_DIR CMake variable to the OpenGL library's installation directory."
    CONDITION QT_FEATURE_gui AND NOT WATCHOS AND ( NOT INPUT_opengl STREQUAL 'no' ) AND NOT QT_FEATURE_opengl_desktop AND NOT QT_FEATURE_opengles2 AND NOT QT_FEATURE_opengl_dynamic
)
```
So they explicitly check against string `"no"`.

Reply:
Oof, that's bad practice. Thanks for checking!